### PR TITLE
docs(README) detailed README

### LINKED ISSUE #14: detail information in README
README is severely lacking information.

It is important for visitors to get a good grasp of what MotorcycleJS is and how it relates/differs from CycleJS.

### PR DESCRIPTION
**Pull request by TylorS**
Update the README to give visitors a sense of what the project is,
how it relates to Cycle.js and why it exists.
Touch on relation to Cycle.js, and that it is not a fork that
disregards it's inspiration entirely.
Introduce the goals of the project, and explain the difference and
similarites in the 'Goals' section.
Add an API section to show that it is the same as Cycle.js

Closes #14

### CODE REVIEW

**Comment by TylorS** [comment]
@staltz I assigned this to you, because it makes the most sense for you to comb through this as it touches on it's relationship to Cycle.js. Feel free to assign it to someone else if you don't have the time.

**Comment by staltz** [comment]
Apart from typos, I really like the README.

File: README.md [hunk 1]
```diff
@@ -1,7 +1,48 @@
...
+$ npm install @motorcycle/core
+```
+
+# Is it a fork?
+
+This is **not** a fork of Cycle.js, but a sister project that will continue to evolve and grow alongside Cycle.js for the forseeable future. All of the core contributors of Motorcycle are, and continue to be, contributors to the Cycle.js project. Advancements in one will bring advancements in the other. Each have similar, yet different goals.
+
+# Goals
+
+The primary focus of this project is to build on the wonderful ideas brought to us by André Medeiros, better known by @staltz, brought to us via Cycle.js and tune it for performance as much a possible.
```
Comment (staltz) [inline README.md]:
```diff
-for performance as much a possible.
+for performance as much as possible.
```

File: README.md [hunk 2]
```diff
@@ -1,7 +1,48 @@
...
+
+ Most.js plays an incredibly huge part in that, as benchmarks place *Most* about 500 times faster than RxJS 4. *Most* is built with a very small core API compared to RxJS and is entirely modular and extensible. For more information on *Most* please visit the github page [here](https://github.com/cujojs/most) and/or join us on gitter [here](https://gitter.im/cujojs/most).
+
+Secondly, the standard DOM Driver, [motorcycle-dom](https://github.com/motorcyclejs/motorcycle-dom), is built using [Snabbdom](https://github.com/paldepind/snabbdom), a more performant virtual-dom implementation than that used by Cycle-DOM. IT continues to implement the same API's, with more performant alterntives to come. Snabbdom eases many issues of using virtual-dom via lifecycle hooks to continue to use existing libraries that work with the DOM directly. Snabbdom is also very modular and can be extended quite easily.
```
Comment (staltz) [inline README.md]:
```diff
-IT continues to implement the same API's,
+It continues to implement the same API's,
```

File: README.md [hunk 3]
```diff
@@ -1,7 +1,48 @@
...
+
+###### Importing
+```js
+//ES6 - recommended =)
+import {run} from '@motorcycle/core'
+//ES5
+var run = require('@motorcycle-core').run
+```
+
+Takes an main function and circularly connects it to the given collection of driver functions.
```
Comment (staltz) [inline README.md]:
```diff
-Takes an main function
+Takes a main function
```

**Comment by TylorS** [comment]
Alright, I'll get to work on these fixes soon.

Any suggestions on clarifying and correcting this?

> The main function expects a collection of "driver source" Observables/Streams as input, and should return a collection of "driver sink" Observables/Streams.

**Comment by Frikki** [comment]
Suggestion:

> The `main` function takes an object of *sources* as input. *Sources* are the outputs from the various drivers. To complete the cycle, `main` should return a *sinks* object, which is a mapping of return values from the program to the drivers, i.e., inputs to the drivers, usually Observables/Streams.

**Comment by Frikki** [comment]
I’d really like to merge this. If there’s no objections, can I reassign this, as André is pretty busy?

**Comment by TylorS** [comment]
All yours @Frikki

**Pull request merged**